dxr/plugins/descriptor/__init__.py

```diff
@@ -48,9 +48,24 @@ class FileToIndex(dxr.indexers.FileToIndex):
     """Do lots of work to yield a description needle."""
 
     comment_re = re.compile(r'(?:.*?/\*+)(?:\s*\*?\s*)(?P<description>.*?)(?:(?:\*+/.*)|(?:$))', flags=re.M)
-    docstring_re = re.compile(r'(?:\'\'\'|""")(?:\s*)(?P<description>.*?)(?:(?:(\'\'\'|"""))|(?:$))', flags=re.M)
+    docstring_re1 = re.compile(r'"""\s*(?P<description>[^"]*)', flags=re.M)
```

Comment:
I tried it, and you're right: you can't repeat a named group on opposite sides of an alternation. Surprising! We could use backreferences…

```python
>>> re.search(r'(foo|bar)A\1', 'barAbar').groups()
('bar',)
>>> re.search(r'(foo|bar)A\1', 'fooAfoo').groups()
('foo',)
```
…to choose the right quotes, but it would get hard to read. Let's go with your way but use a `for` loop so we don't have to worry about having/testing both branches of code.